cmd/sql/flow.go

```diff
@@ -4,273 +4,235 @@ import (
+		}
+	case configCmdName:
+		if projectDir != "" {
+			projectDir = resolvePath(projectDir)
+			args = append(args, "--project-dir", projectDir)
+		}
+		if env != "" {
+			args = append(args, "--env", env)
+		}
+	case validateCmdName:
+		if env != "" {
+			args = append(args, "--env", env)
+		}
+		if connection != "" {
+			args = append(args, "--connection", connection)
+		}
+	case generateCmdName:
+		if projectDir != "" {
+			projectDir = resolvePath(projectDir)
+			args = append(args, "--project-dir", projectDir)
+		}
+		if env != "" {
+			args = append(args, "--env", env)
+		}
+		if generateTasks {
+			args = append(args, "--generate-tasks")
+		}
+		if noGenerateTasks {
+			args = append(args, "--no-generate-tasks")
+		}
+	case runCmdName:
+		if projectDir != "" {
+			projectDir = resolvePath(projectDir)
+			args = append(args, "--project-dir", projectDir)
+		}
+		if env != "" {
+			args = append(args, "--env", env)
+		}
+		if generateTasks {
+			args = append(args, "--generate-tasks")
+		}
+		if noGenerateTasks {
+			args = append(args, "--no-generate-tasks")
+		}
+		if verbose {
+			args = append(args, "--verbose")
 		}
 	}
+	return args
+}
 
-	if setAirflowHome && airflowHome != "" {
-		airflowHomeAbs, err := getAbsolutePath(airflowHome)
-		if err != nil {
-			return nil, nil, err
+// Create directories and resolve their filepaths to absolute
+func createDirs(cmd *cobra.Command) []string {
+	var paths []string
+	switch cmd.Name() {
+	case initCmdName:
+		if projectDir != "" {
+			paths = append(paths, projectDir)
+		}
+		if airflowHome != "" {
+			paths = append(paths, airflowHome)
 		}
-		flags["airflow-home"] = airflowHomeAbs
-		mountDirs = append(mountDirs, airflowHomeAbs)
+		if airflowDagsFolder != "" {
+			paths = append(paths, airflowDagsFolder)
+		}
+	case configCmdName:
+		if projectDir != "" {
+			paths = append(paths, projectDir)
+		}
+	case validateCmdName:
+		if projectDir != "" {
+			paths = append(paths, projectDir)
+		}
+		paths = append(paths, readConfigCmdOutput("airflow_home"))
+	case generateCmdName:
+		if projectDir != "" {
+			paths = append(paths, projectDir)
+		}
+		paths = append(paths, readConfigCmdOutput("airflow_home"), readConfigCmdOutput("airflow_dags_folder"))
+	case runCmdName:
+		if projectDir != "" {
+			paths = append(paths, projectDir)
+		}
+		paths = append(paths, readConfigCmdOutput("airflow_home"), readConfigCmdOutput("airflow_dags_folder"))
 	}
-
-	if setAirflowDagsFolder && airflowDagsFolder != "" {
-		airflowDagsFolderAbs, err := getAbsolutePath(airflowDagsFolder)
-		if err != nil {
-			return nil, nil, err
+	mounts := make([]string, len(paths))
+	for i, path := range paths {
+		absPath := resolvePath(path)
+		if err := os.MkdirAll(absPath, os.ModePerm); err != nil {
+			panic(err)
 		}
-		flags["airflow-dags-folder"] = airflowDagsFolderAbs
-		mountDirs = append(mountDirs, airflowDagsFolderAbs)
+		mounts[i] = absPath
 	}
-
-	return flags, mountDirs, nil
+	return mounts
```

Comment:
Did we have a ticket somewhere saying we're mounting duplicate directories. Should we check here that mounts have unique values and no duplicates?

Reply:
Yes, I created https://github.com/astronomer/astro-sdk/issues/1457 for this.

Reply:
Fixed in fbbefd0